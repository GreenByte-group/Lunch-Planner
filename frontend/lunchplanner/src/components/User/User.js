import React from "react"
import moment from "moment"
import ListItem from "material-ui/List/ListItem";
import {withStyles} from "material-ui";
import {Schedule, Today} from "@material-ui/icons";
import AcceptedButton from "../Event/AcceptedButton";

const styles = {
    listItem: {
        padding: '20px 16px',
        '&:hover': {
            backgroundColor: '#0303031a !important',
        },
        transition: 'background 0.5s',
    },
    content: {
        width: '100%',
    },
    selected: {
        backgroundColor: '#bacfa255',
        '&:hover': {
            backgroundColor: '#bacfa288 !important',
        }
    },
    text: {
        marginLeft: '24px',
        marginTop: '6px',
        float: 'left',
    },
    username: {
        fontFamily: "Work Sans",
        fontSize: '13px',
        lineHeight: '20px',
    },
    placeholder: {
        height: '100%',
        display: 'inline-block',
        verticalAlign: 'middle',
    },
    profilePicture: {
        float: 'left',
        border: '1px black solid',
        borderRadius: '50%',
        height: '32px',
        width: '32px',
    }
};

class Event extends React.Component {

    constructor(props) {
        super();

        this.state = {
            selected: props.selected | false,
            username: props.username,
        };
    }

    componentWillReceiveProps(newProps) {
        if(newProps.username && newProps.username !== this.state.username) {
            console.log('username: ' + newProps.username);
            this.setState ({
                username: newProps.username,
            });
        }

        if(newProps.selected !== undefined && newProps.selected !== null && newProps.selected !== this.state.selected) {
            this.setState ({
                selected: newProps.selected,
            });
        }
    }

    clickHandler = () => {
        this.props.onClick(this.state.username, !this.state.selected);

        this.setState({
            selected: !this.state.selected,
        });
    };

    render() {
        const {classes} = this.props;
        let selected= this.state.selected;
        let username = this.state.username;

        let listClasses = classes.listItem;
        if(selected)
            listClasses += " " + classes.selected;

        return (
            <ListItem button className={listClasses} onClick={this.clickHandler}>
                <div className={classes.content}>
                    {/*TODO picture*/}
                    <div className={classes.profilePicture}></div>
                    <div className={classes.text}>
                        <span className={classes.username}>{username}</span>
                    </div>
                </div>
                {(selected
                        ? <AcceptedButton />
                        : ""
                )}
            </ListItem>
        );
    }
}

export default withStyles(styles)(Event);