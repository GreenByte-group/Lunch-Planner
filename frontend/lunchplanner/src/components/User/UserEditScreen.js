import React from 'react';
import {withStyles} from "@material-ui/core/styles/index";
import {TextField, FormControl, InputLabel, Input, Button} from "@material-ui/core";
import {Edit} from "@material-ui/icons";
import {getUsername} from "../authentication/LoginFunctions";
import {
    deleteAccount,
    getProfilePicturePath,
    getUser,
    updateEmail,
    updatePassword,
    updateProfilePicture
} from "./UserFunctions";
import {HOST} from "../../Config";
import {userNeedReload} from "../AppContainer";

const styles = theme => ({
    root: {
        width: '100%',
        height: '100%',
        overflowY: 'auto',
        maxWidth: '1024px',
        marginLeft: 'auto',
        marginRight: 'auto',
        display: 'flex',
        flexDirection: 'column',
        paddingTop: '20px',
        paddingLeft: '10px',
        paddingRight: '10px',
    },
    pictureName: {
        display: 'flex',
        flexDirection: 'row',
        alignItems: 'center',
    },
    profilePicture: {
        width: '200px',
        height: '200px',
        border: '1px solid black',
        borderRadius: '50%',
        float: 'left',
        backgroundPosition: 'center',
        backgroundSize: 'cover',
        overflow: 'hidden',
    },
    profilePictureImg: {
        objectFit: 'cover',
        width: '100%',
        height: '100%',
    },
    editIconWrapper: {
        width: '200px',
        height: '200px',
        borderRadius: '50%',
        backgroundColor: 'rgba(0,0,0,0.3)',
        transition: 'background-color 0.5s',
        "&:hover": {
            backgroundColor: 'rgba(0,0,0,0.5)',
            cursor: 'pointer',
        }
    },
    editIcon: {
        width: '30px',
        height: '30px',
        marginLeft: '85px',
        marginTop: '85px',
        color: 'white',
    },
    username: {
        marginLeft: '25px',
        fontSize: '20px',
        fontWeight: 600,
    },
    input: {
        display: 'none !important',
    },
    inputLabel: {
        height: '200px',
        width: '200px',
        margin: 0,
        padding: 0,
    },
    edit: {
        marginTop: '20px',
        display: 'flex',
        flexDirection: 'column',
    },
    editArea: {
        display: 'flex',
        flexDirection: 'row',
        marginBottom: '15px',
    },
    text: {
        width: '200px',
    },
    textFieldMail: {
        marginLeft: '5px',
        width: '100%',
        maxWidth: '300px',
    },
    button: {
        boxShadow: '0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)',
        backgroundColor: '#75a045',
        color: 'white',
        "&:hover": {
            backgroundColor: '#9cd556',
        }
    },
    error: {
        color: 'red',
    },
    placeGiver: {
        padding: '10px',
    }
});

class UserEditScreen extends React.Component {

    constructor(props) {
        super();

        this.state = {
            email: '',
            showPassword: false,
            password: '',
            repeat: '',
            emailError: null,
            passwordError: null,
            pathImage: null,
            delete: false,
            pathProfilePicture: "",
        };

        this.getUser();
    }

    getUser = () => {
        getUser(getUsername(), (response) => {
            this.setState({
                pathProfilePicture: response.data.profilePictureUrl,
                email: response.data.eMail,
            })
        })
    }

    onChange = (value) => {
        this.setState({
            [value.target.id]: value.target.value,
        })
    };

    onProfile = (event) => {
        let fileList = event.target.files;
        //fileList.item(0);
        this.setState({
            pathImage: fileList.item(0),
        })
    };


    onDeleteClick = () => {
        this.setState({
            delete: true,
        });
        deleteAccount(getUsername());
    };

    onSubmit = () => {
        if(this.state.pathImage) {
            updateProfilePicture(this.state.pathImage, (response) => {
                userNeedReload();
            })
        }

        if(this.state.email) {
            updateEmail(this.state.email, (response) => {

                if(response.status !== 204) {
                    this.setState({emailError: response.response.data})
                    userNeedReload();
                } else {
                    this.setState({
                        emailError: null,
                        email: '',
                    })
                }
            });
        }

        if(this.state.password && this.state.password === this.state.repeat) {
            updatePassword(this.state.password, (response) => {
                if(response.status !== 204) {
                    this.setState({passwordError: response.response.data})
                } else {
                    this.setState({
                        passwordError: null,
                        password: '',
                        repeat: '',
                    })
                }
            })
        } else if (this.state.password !== this.state.repeat){
            this.setState({passwordError: 'Passwords are different'})
        }
    };

    render() {
        const { classes } = this.props;

        let url;
        if(this.state.pathImage) {
            console.log('Profile picture', this.state.pathImage);
            console.log('picturepath', this.state.pathProfilePicture)
            url = URL.createObjectURL(this.state.pathImage)
        } else {
            url = HOST + this.state.pathProfilePicture;
        }

        return (
            <div className={classes.root}>
                <div  className={classes.pictureName}>
                    <div
                        ref="image-pane"
                        className={classes.profilePicture}
                    >
                        <label htmlFor="file" className={classes.inputLabel}>
                            <div className={classes.editIconWrapper} >
                                <Edit className={classes.editIcon} />
                            </div>

                            <img src={url} className={classes.profilePictureImg} />
                        </label>
                        <input className={classes.input} type="file" accept="image/*" id="file" onChange={this.onProfile} />
                    </div>
                    <div className={classes.username}>
                        {getUsername()}
                    </div>
                </div>
                <div className={classes.edit}>
                    {
                        (this.state.emailError)
                            ? <p className={classes.error}>{this.state.emailError}</p>
                            : ''
                    }
                    {
                        (this.state.passwordError)
                            ? <p className={classes.error}>{this.state.passwordError}</p>
                            : ''
                    }
                    <div className={classes.editArea}>
                        <TextField className={classes.textFieldMail}
                                   label="E-Mail"
                                   id="email"
                                   value={this.state.email}
                                   onChange={this.onChange}
                        />
                    </div>
                    { (this.state.showPassword) ?
                        <div>
                            <div className={classes.editArea}>
                                <TextField className={classes.textFieldMail}
                                           type="password"
                                           label="New password"
                                           id="password"
                                           value={this.state.password}
                                           onChange={this.onChange}
                                />
                            </div>
                            <div className={classes.editArea}>
                                <TextField className={classes.textFieldMail}
                                           label="Repeat password"
                                           type="password"
                                           id="repeat"
                                           value={this.state.repeat}
                                           onChange={this.onChange}
                                />
                            </div>
                        </div>
                        :
                        <div className={classes.editArea}>
                            <Button
                                className={classes.button}
                                onClick={() => this.setState({showPassword: true})}
                                variant="contained"
                                color="primary"
                            >
                                Change password
                            </Button>
                            <p className={classes.placeGiver}/>
                            <Button
                                className={classes.button}
                                onClick={this.onDeleteClick}
                                variant="contained"
                                color="primary"
                            >
                                Delete my account
                            </Button>
                        </div>
                    }
                    <div className={classes.editArea}>
                        <Button
                            className={classes.button}
                            onClick={this.onSubmit}
                            variant="contained"
                            color="primary"
                        >
                            Save
                        </Button>
                    </div>
                </div>
            </div>
        )
    }
}

export default withStyles(styles, { withTheme: true })(UserEditScreen);