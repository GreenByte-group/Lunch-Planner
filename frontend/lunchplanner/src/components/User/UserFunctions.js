import axios from "axios";
import {HOST} from "../../Config";

export function getUsers(search, responseFunc) {
    let url;
    if(search == null || search === undefined || search === "")
        url = HOST + "/user";
    else
        url = HOST + "/user/search/" + search;

    axios.get(url)
        .then(responseFunc);
}


export function getUser(username, responseFunc) {
    let url = HOST + "/user/" + username;
    axios.get(url)
        .then(responseFunc);
}

export function inviteExtern(eventId, responseFunc) {
    let url = HOST + "/event/" + eventId +  "/token";
    axios.get(url)
        .then(responseFunc);
}

export function updateEmail(email, responseFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + "/user/options/profile/mail";
    axios.put(url, email, config)
        .then(responseFunc)
        .catch(responseFunc);
}

export function updatePassword(password, responseFunc) {
    let config = {
        headers: {
            'Content-Type': 'text/plain',
        }
    };

    let url = HOST + "/user/options/profile/password";
    axios.put(url, password, config)
        .then(responseFunc)
        .catch(responseFunc);
}

export function updateProfilePicture(file, responseFunc) {
    let config = {
        headers:{
            'Content-Type':'multipart/form-data'
        }
    };

    const formData = new FormData();
    formData.append('file',file);

    let url = HOST + "/user/options/profile/picture/upload";
    axios.post(url, formData, config)
        .then(responseFunc)
        .catch(responseFunc);
}

export function getProfilePicturePath(username, responseFunc) {
    let url = HOST + "/user/getProfilePicture/" + username;
    axios.get(url)
        .then(responseFunc);
}