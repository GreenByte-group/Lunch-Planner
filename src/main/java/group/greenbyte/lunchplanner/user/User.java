package group.greenbyte.lunchplanner.user;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class User {

    @Id
    @Column(nullable = false, length = 50)
    private String userName;

    @Column(nullable = false, length = 50)
    private String eMail;

    @Column(nullable = false, length = 80)
    private String password;

    /*
    @ManyToMany(mappedBy = "usersInvited")
    private Set<Event> eventsInvited = new HashSet<>();

    @ManyToMany(mappedBy = "usersAdmin")
    private Set<Event> eventsAdmin = new HashSet<>();
    */

}
