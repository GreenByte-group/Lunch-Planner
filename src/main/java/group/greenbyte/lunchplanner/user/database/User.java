package group.greenbyte.lunchplanner.user.database;

import group.greenbyte.lunchplanner.event.database.Comment;
import group.greenbyte.lunchplanner.event.database.EventInvitation;
import group.greenbyte.lunchplanner.team.database.TeamMember;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

@Entity
public class User {

    static final public int MAX_USERNAME_LENGTH = 50;
    static final public int MAX_MAIL_LENGTH = 50;
    static final public int MAX_PASSWORD_LENGTH = 200;

    @Id
    @Column(unique = true,length = MAX_USERNAME_LENGTH)
    private String userName;

    @Column(nullable = false, length = MAX_MAIL_LENGTH)
    private String eMail;

    @Column(nullable = false, length = MAX_PASSWORD_LENGTH)
    private String password;

    @Column
    private String token;

    @OneToMany(mappedBy = "eventInvited", cascade = CascadeType.ALL)
    private Set<EventInvitation> eventsInvited;

    @OneToMany(mappedBy = "team", cascade = CascadeType.ALL)
    private Set<TeamMember> teamsMember = new HashSet<>();

    @OneToMany(mappedBy = "eventComment", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
    private List<Comment> comments = new ArrayList<>();

    public String getUserName() {
        return userName;
    }

    public void setUserName(String userName) {
        this.userName = userName;
    }

    public String geteMail() {
        return eMail;
    }

    public void seteMail(String eMail) {
        this.eMail = eMail;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public Set<EventInvitation> getEventsInvited() {
        return eventsInvited;
    }

    public void setEventsInvited(Set<EventInvitation> eventsInvited) {
        this.eventsInvited = eventsInvited;
    }

    public void addEventsInvited(EventInvitation eventInvitation) {
        if(eventsInvited == null)
            eventsInvited = new HashSet<>();

        eventsInvited.add(eventInvitation);
    }

    public String getToken() {
        return token;
    }

    public void setToken(String token) {
        this.token = token;
    }
}
