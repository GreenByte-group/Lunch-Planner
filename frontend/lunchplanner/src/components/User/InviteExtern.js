import React from 'react';
import PropTypes from "prop-types";
import {withStyles} from "@material-ui/core/styles/index";
// import CopyButton from "@material-ui/icons/"
import {inviteExtern} from "./UserFunctions";
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import Input from '@material-ui/core/Input';
import {getHistory} from "../../utils/HistoryUtils";
import {FRONTEND_HOST} from "../../Config";
import IconButton from '@material-ui/core/IconButton';
import {CopyToClipboard} from 'react-copy-to-clipboard';

const styles = {
    linkField:{
        marginLeft: 24,
        width: '300px',

    }
};

class InviteExtern extends React.Component {

    constructor(props) {
        super();
        let eventId = props.match.params.eventId;
        let inviteLink = inviteExtern(eventId, (response) =>{
            let link = FRONTEND_HOST + "/public/" + response.data;
            this.setState({
                link: link,
            });
        });
        this.state = {
            open: true,
            link: "",
            copied: false,
            eventId: eventId,
        };
    }

    handleClose = () => {
        this.setState({ open: false });
        getHistory().push(this.props.location.query.source);
    };

    handleCopy = () =>{
        this.setState({
            copied: true,
        })
    };


    render(){

        const classes = this.props;
        let link = this.state.link;

        return(
            <div>
            <Dialog
                open={this.state.open}
                closeUrl={"/app/event/" + this.state.eventId}
            >
                <DialogTitle id="alert-dialog-title">{"Share this event with other people"}</DialogTitle>
                <DialogContent>
                    <DialogContentText >
                        Copy this link an invite other people per e-mail
                        <br/>
                        Link: <Input value={link}/>
                        <CopyToClipboard text={link}
                                         onCopy={this.handleCopy}>
                            <IconButton>
                                <p>COPY</p>
                            </IconButton>
                        </CopyToClipboard>


                    </DialogContentText>
                </DialogContent>
                <DialogActions>
                    <Button onClick={this.handleClose} color="primary">
                        Close
                    </Button>
                </DialogActions>
            </Dialog>
        </div>);
    }
}

InviteExtern.propTypes = {
    classes: PropTypes.object.isRequired,
};
export default withStyles(styles, { withTheme: true })(InviteExtern);