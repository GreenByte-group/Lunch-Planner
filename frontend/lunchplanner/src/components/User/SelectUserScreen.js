import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from 'material-ui/styles';
import Dialog from 'material-ui/Dialog';
import AppBar from 'material-ui/AppBar';
import Toolbar from 'material-ui/Toolbar';
import IconButton from 'material-ui/IconButton';
import Typography from 'material-ui/Typography';
import CloseIcon from '@material-ui/icons/Close';
import Slide from 'material-ui/transitions/Slide';

import {Link} from "react-router-dom";
import axios from "axios/index";
import {HOST} from "../../Config";
import {List} from "material-ui";
import User from "./User";
import FloatingActionButton from "../FloatingActionButton";

const styles = {
    root: {
      position: 'fixed'
    },
    appBar: {
        position: 'fixed',
    },
    flex: {
        flex: 1,
    },
    list: {
        marginTop: '56px',
        padding: 0,
    },

};

function Transition(props) {
    return <Slide direction="up" {...props} />;
}

class SelectUserScreen extends React.Component {

    constructor(props) {
        super();

        let array;
        if(props.location.query && props.location.query.invitedUsers) {
            let string = String(props.location.query.invitedUsers);
            if(string !== "")
                array = string.split(',');
        }

        this.state = {
            open: true,
            users: [],
            selectedUsers: array || [],
        };
    }

    componentDidMount() {
        this.setState({
            search: this.props.search,
        });

        let url;
        if(this.props.search)
            url = HOST + "/user/search/" + this.props.search;
        else
            url = HOST + "/user";

        axios.get(url)
            .then((response) => {
                this.setState({
                    users: response.data,
                })
            });

        // this.setState({
        //     users: [{username: "Martin"}, {username: "Sarah"}, {username: "Test"}, {username: "Test2"}]
        // });
    }

    clickHandler = (username, selected) => {
        let selectedUsers = this.state.selectedUsers;
        if(selected) {
            selectedUsers.push(username);
        } else {
            let index = selectedUsers.indexOf(username);
            selectedUsers.splice(index, 1);
        }

        this.setState({
            selectedUsers: selectedUsers,
        });
    };

    handleSend = () => {
        if(this.props.location.query)
            this.props.history.push(this.props.location.query.source +
                "?invitedUsers=" + this.state.selectedUsers);
    };

    render() {
        const { classes } = this.props;
        let users = this.state.users;
        let countSelected = this.state.selectedUsers.length;
        let textTitle = "Select users";
        if(countSelected !== 0) {
            textTitle = countSelected + " selected";
        }

        return (
            <div className={classes.root}>
                <Dialog
                    fullScreen
                    open={this.state.open}
                    onClose={this.handleClose}
                    transition={Transition}
                >
                    <AppBar className={classes.appBar} color ="white">
                        <Toolbar>
                            <Link to="/event/create">
                                <IconButton color="inherit" aria-label="Close">
                                    <CloseIcon />
                                </IconButton>
                            </Link>
                            <Typography variant="title" color="inherit" className={classes.flex}>
                                {textTitle}
                            </Typography>

                        </Toolbar>
                    </AppBar>
                    <List className={classes.list}>
                        {users.map((listValue) => {
                            return <User selected={this.state.selectedUsers.includes(listValue.userName)} username={listValue.userName} onClick={this.clickHandler}/>;
                        })}
                    </List>
                    <FloatingActionButton onClick={this.handleSend} icon="done"/>
                </Dialog>
            </div>
        );
    }
}

SelectUserScreen.propTypes = {
    classes: PropTypes.object.isRequired,
};
export default withStyles(styles, { withTheme: true })(SelectUserScreen);