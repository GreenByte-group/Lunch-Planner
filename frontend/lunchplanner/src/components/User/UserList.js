import React from 'react';
import {withStyles} from 'material-ui/styles';
import {List} from "material-ui";
import User from "./User";

const styles = {
    list: {
        padding: 0,
        marginTop: '0px !important',
    },
};

class UserList extends React.Component {

    constructor(props) {
        super();

        this.state = {
            selectedUsers: props.selectedUsers || [],
            users: props.users || [],
            selectable: props.selectable || false,
            onSelectionChanged: props.onSelectionChanged,
            othersInvited: props.othersInvited || false,
        };
    }

    componentWillReceiveProps(newProps) {
        let selectedUsers, users;

        if(newProps.selectedUsers && newProps.selectedUsers !== this.state.selectedUsers) {
            selectedUsers = newProps.selectedUsers;
            this.setState({
                selectedUsers: selectedUsers,
            });
        }

        if(newProps.users && newProps.users !== this.state.users) {
            users = newProps.users;
            this.setState({
                users: users,
            });
        }
    }

    clickHandler = (username, selected) => {
        if(this.state.selectable) {
            let selectedUsers = this.state.selectedUsers;
            if (selected) {
                selectedUsers.push(username);
            } else {
                let index = selectedUsers.indexOf(username);
                selectedUsers.splice(index, 1);
            }

            this.setState({
                selectedUsers: selectedUsers,
            });

            this.state.onSelectionChanged(selectedUsers);
        }
    };

    render() {
        const { classes } = this.props;
        let users = this.state.users;
        let selectedUsers = this.state.selectedUsers;
        let othersInvited = this.state.othersInvited;

        return (
            <List
                className={classes.list}
            >
                {users.map((listValue) => {
                    return <User invited={othersInvited} selectable={this.state.selectable} selected={selectedUsers.includes(listValue.userName)} username={listValue.userName} onClick={this.clickHandler}/>;
                })}
            </List>
        );
    }
}

export default withStyles(styles, { withTheme: true })(UserList);