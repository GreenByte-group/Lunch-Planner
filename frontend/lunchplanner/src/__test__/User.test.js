import React from 'react';
import User from '../components/User/User';
import renderer from 'react-test-renderer';
import {MemoryRouter} from "react-router-dom";

it ("renders correctly", ()=>{
    const tree = renderer.create(
        <User/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
});