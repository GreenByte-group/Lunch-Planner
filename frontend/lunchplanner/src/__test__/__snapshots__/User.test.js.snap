// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiButtonBase-root-19 MuiListItem-root-9 MuiListItem-default-12 MuiListItem-gutters-16 MuiListItem-button-17 Event-listItem-1 Event-unselectable-4"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  <div
    className="Event-content-2"
  >
    <div
      className="Event-profilePicture-8"
    />
    <div
      className="Event-text-5"
    >
      <span
        className="Event-username-6"
      />
    </div>
  </div>
  
  
  <span
    className="MuiTouchRipple-root-22"
  />
</div>
`;
